network-api/networkapi/wagtailpages/tests/buyersguide/test_taxonomies.py

```diff
@@ -1,71 +1,69 @@
             slug="not-the-slugified-title",
         )
 
-        category.save()
+        category.clean()
 
-        self.assertEqual(category.title, "Test category")
-        self.assertEqual(category.slug, "not-the-slugified-title")
+        self.assertEqual(category.title, TEST_CATEGORY_TITLE)
+        self.assertEqual(category.slug, "en-test-category")
 
-    def test_slug_has_to_be_unique(self):
+    def test_clean_raises_if_category_with_same_title_and_locale_in_db(self):
         buyersguide_factories.BuyersGuideContentCategoryFactory(
-            title="Test category",
-            slug="test-category",
+            title=TEST_CATEGORY_TITLE,
         )
         category = buyersguide_factories.BuyersGuideContentCategoryFactory.build(
-            title="Test category",
-            slug="test-category",
+            title=TEST_CATEGORY_TITLE,
         )
 
         with self.assertRaises(exceptions.ValidationError):
-            category.full_clean()
-
-    @unittest.skip(
-        "We would really want the slugs to be only unique per locale. "
-        "If we implement that with a UniqueContraint or unique_together, "
-        "Wagtail will crash if the constraint is violated. "
-        "Wagtail does handle the simple unique requirement on the slug field gracefully. "
-        "Therefore, we are using slugs that are unique regardless of locale. "
-        "We need to manually create unique slugs if there are clashes between the locales. "
-        "Because we are using slugs that need to be unique regradless of locale, this "
-        "test would fail. Therefore it is skipped."
-        "See also: https://github.com/wagtail/wagtail/issues/8918"
-    )
-    def test_same_slug_allowed_on_different_locale(self):
-        category_default_locale = buyersguide_factories.BuyersGuideContentCategoryFactory(
-            title="Test category",
-            slug="test-category",
-            locale=self.default_locale,
+            category.clean_slug()
+
+    def test_clean_slug_different_for_different_locales(self):
+        category_default_locale = buyersguide_factories.BuyersGuideContentCategoryFactory.build(
+            title=TEST_CATEGORY_TITLE,
         )
         category_fr_locale = buyersguide_factories.BuyersGuideContentCategoryFactory.build(
-            title="Test category",
-            slug="test-category",
+            title=TEST_CATEGORY_TITLE,
             locale=self.fr_locale,
         )
 
-        category_fr_locale.save()
+        category_default_locale.clean_slug()
+        category_fr_locale.clean_slug()
 
-        self.assertEqual(category_fr_locale.slug, category_default_locale.slug)
+        self.assertEqual(category_fr_locale.title, category_default_locale.title)
         self.assertNotEqual(category_fr_locale.locale, category_default_locale.locale)
+        self.assertNotEqual(category_fr_locale.slug, category_default_locale.slug)
+
+    def test_can_clean_and_save_copy_for_translation(self):
```

Comment:
Look like this test is just testing clean and save together but with no expected assertions. I'd suggest it's worth throwing some expectation in there, like asserting the saved objects slug is what you'd expect to have been stored.

Reply:
Good call.